tests/db/test_mysql.py

```diff
@@ -148,56 +153,143 @@ def test_partition():
+    reset_is_current_flags(cursor, ["rna_A01", "rna_A03"])
+    connection.commit()
+
+    cursor.execute("SELECT root_sample_id FROM lighthouse_sample WHERE is_current=0;")
+    rows = [row[0] for row in cursor.fetchall()]
+    assert rows == ["rna_1", "rna_2", "rna_3", "rna_5"]
+
+    cursor.execute("SELECT root_sample_id FROM lighthouse_sample WHERE is_current=1;")
+    rows = [row[0] for row in cursor.fetchall()]
+    assert rows == ["rna_4"]
+
+    cursor.execute("SELECT DISTINCT(updated_at) FROM lighthouse_sample;")
+    assert len(cursor.fetchall()) == 2
+
+
+def test_insert_samples_in_mlwh_inserts_one_complete_sample_correctly(config, mlwh_rw_db, logging_messages):
+    _, cursor = mlwh_rw_db
+
+    with patch("crawler.db.mysql.map_mongo_sample_to_mysql"):
+        with patch("crawler.db.mysql.set_is_current_on_mysql_samples") as make_mysql_samples:
+            make_mysql_samples.return_value = [MLWH_SAMPLE_COMPLETE]
+            insert_or_update_samples_in_mlwh([{"pseudo": "sample"}], config, LoggingCollection(), logging_messages)
+
+    fields = [
+        "ch1_cq",
+        "ch1_result",
+        "ch1_target",
+        "ch2_cq",
+        "ch2_result",
+        "ch2_target",
+        "ch3_cq",
+        "ch3_result",
+        "ch3_target",
+        "ch4_cq",
+        "ch4_result",
+        "ch4_target",
+        "coordinate",
+        "date_tested",
+        "filtered_positive",
+        "filtered_positive_timestamp",
+        "filtered_positive_version",
+        "is_current",
+        "lab_id",
+        "lh_sample_uuid",
+        "lh_source_plate_uuid",
+        "mongodb_id",
+        "must_sequence",
+        "plate_barcode",
+        "preferentially_sequence",
+        "result",
+        "rna_id",
+        "root_sample_id",
+        "source",
+    ]
+    cursor.execute(f"SELECT {','.join(fields)} FROM lighthouse_sample;")
+    rows = [row for row in cursor.fetchall()]
+    assert len(rows) == 1
+
+    row = rows[0]
+    assert row == (
+        Decimal("24.67"),
+        "Positive",
+        "A gene",
+        Decimal("23.92"),
+        "Negative",
+        "B gene",
+        Decimal("25.12"),
+        "Positive",
+        "C gene",
+        Decimal("22.86"),
+        "Negative",
+        "D gene",
+        "C3",
+        datetime(2021, 2, 3, 4, 5, 6),
+        True,
+        datetime(2021, 2, 3, 5, 6, 7),
+        "v3",
+        True,
+        "BB",
+        "233223d5-9015-4646-add0-f358ff2688c7",
+        "c6410270-5cbf-4233-a8d1-b08445bbac5e",
+        "6140f388800f8fe309689124",
+        True,
+        "95123456789012345",
+        False,
+        "Positive",
+        "95123456789012345_C03",
+        "BAA94123456",
+        "Bob's Biotech",
+    )
 
-        cursor.execute("SELECT root_sample_id FROM lighthouse_sample WHERE is_current=1;")
-        rows = [row[0] for row in cursor.fetchall()]
-        assert rows == ["rna_4"]
 
-    finally:
-        cursor.execute("DELETE FROM lighthouse_sample;")
-        connection.close()
```

Comment:
This block of code was to reset the testing database to where it was before the test. We haven't refactored this to go into a fixture, so this is needed in every test that adds data to the lighthouse_sample table.

Reply:
It is in the fixture.  The cursor gets its connection from another fixture which is also set up to truncate the table.

Reply:
Ok, I see it now in mlwh_rw_db, Thanks